gluten-core/src/main/java/io/glutenproject/substrait/expression/ExpressionBuilder.java

```diff
@@ -199,6 +194,23 @@ public static ExpressionNode makeLiteral(Object obj, DataType dataType, Boolean
       } else {
         return makeDecimalLiteral((Decimal) obj);
       }
+    }  else if (dataType instanceof ArrayType) {
+      if (obj == null) {
+        return makeNullLiteral(TypeBuilder.makeDecimal(nullable, 0, 0));
+      } else {
+        Object[] arrays = ((GenericArrayData) obj).array();
+        Arrays.stream(arrays).collect(Collectors.toList());
+        ArrayList<String> list = new ArrayList<>();
+        for (Object array : arrays) {
+          if (array instanceof UTF8String) {
+            list.add(array.toString());
+          } else {
+            throw new UnsupportedOperationException(
+                    String.format("Type not supported: %s.", dataType.toString()));
+          }
+        }
+        return makeListLiteral(list);
```

Comment:
@zhejiangxiaomai Would you like to help review the list literal part here?

Reply:
Hi @lviiii, there are already some functions to create List literal such as makeStringListLiteral, makeDoubleListLiteral etc., would you like to take a look?

https://github.com/oap-project/gluten/blob/main/gluten-core/src/main/java/io/glutenproject/substrait/expression/ExpressionBuilder.java#L115-L117

Reply:
I've removed the redundant code.